website/addons/googledrive/model.py

```diff
@@ -2,235 +2,189 @@
-            return self.oauth_settings.fetch_access_token(force_refresh)
-        return None
-
-    def clear(self):
-        if self.oauth_settings:
-            self.oauth_settings.revoke_access_token()
-            self.oauth_settings = None
-            self.save()
-
-        for node_settings in self.googledrivenodesettings__authorized:
-            node_settings.deauthorize(Auth(self.owner))
-            node_settings.save()
-
-    def save(self, *args, **kwargs):
-        if self.oauth_settings:
-            self.oauth_settings.save()
-        return super(GoogleDriveUserSettings, self).save(*args, **kwargs)
-
-    def delete(self, save=True):
-        self.clear()
-        super(GoogleDriveUserSettings, self).delete(save)
-
-    def __repr__(self):
-        return u'<GoogleDriveUserSettings(user={self.owner.username!r})>'.format(self=self)
-
-
-class GoogleDriveNodeSettings(StorageAddonBase, AddonNodeSettingsBase):
-
-    folder_id = fields.StringField(default=None)
-    folder_path = fields.StringField(default=None)
-
-    user_settings = fields.ForeignField(
-        'googledriveusersettings', backref='authorized'
-    )
+    def complete(self):
+        return bool(self.has_auth and self.user_settings.verify_oauth_access(
+            node=self.owner,
+            external_account=self.external_account,
+            metadata={'folder': self.drive_folder_id}
+        ))
 
-    @property
     def folder_name(self):
-        if not self.folder_id:
+        if not self.drive_folder_id:
             return None
 
         if self.folder_path != '/':
-            return unquote(os.path.split(self.folder_path)[1].encode('utf-8')).decode('utf-8')
-
-        return '/ (Full Google Drive)'
+            # `urllib` does not properly handle unicode.
+            # encode input to `str`, decode output back to `unicode`
+            return urllib.unquote(os.path.split(self.folder_path)[1].encode('utf-8')).decode('utf-8')
 
-    @property
-    def has_auth(self):
-        """Whether an access token is associated with this node."""
-        return bool(self.user_settings and self.user_settings.has_auth)
-
-    @property
-    def complete(self):
-        return self.has_auth and self.folder_id is not None
-
-    def deauthorize(self, auth=None, add_log=True):
-        """Remove user authorization from this node and log the event."""
-        if add_log:
-            extra = {'folder': self.folder_name}
-            nodelogger = GoogleDriveNodeLogger(node=self.owner, auth=auth)
-            nodelogger.log(action="node_deauthorized", extra=extra, save=True)
-
-        self.folder_id = None
+    def clear_auth(self):
+        self.drive_folder_id = None
         self.folder_path = None
-        self.user_settings = None
+        self.drive_folder_name = None
+        return super(GoogleDriveNodeSettings, self).clear_auth()
 
-        self.save()
+    def set_auth(self, *args, **kwargs):
+        self.drive_folder_id = None
```

Comment:
`set_auth` is only called when a user imports an access token to associate it with a node. If it was previously authorized, then deauthorized and not cleared properly, it would already have the previously selected folder displayed as the selected folder before a user made that selection for this token import. This shouldn't ever happen, but I think this line was included just to be safe.